src/test/ui/inference/deref-suggestion.stderr

```diff
@@ -7,6 +7,8 @@ LL |     foo(s);
    |     |   expected struct `String`, found `&String`
    |     arguments to this function are incorrect
    |
+   = note: expected struct `String`
+           found reference `&String`
```

Comment:
This PR changed a lot of the UI tests to have duplicated-ish note messages. I don't know enough about the diagnostics system to really understand why.

(Several similar changes, e.g. `expected struct String, found type str` in [src/test/ui/issues/issue-53348.stderr](https://github.com/rust-lang/rust/pull/101551/commits/a2b102a9480175cb84ee5e0e0eab2073e09b5295#diff-b50b2ccce45b27f4f4af0e4253b4a343ab304f0b2ede0b08b39f56731567fdcb), `expected struct String, found unit type ()` in [src/test/ui/block-result/consider-removing-last-semi.stderr](https://github.com/rust-lang/rust/pull/101551/commits/a2b102a9480175cb84ee5e0e0eab2073e09b5295#diff-5f0671dff1f32b59ad63201cdf56f12b1ba3e78e9d21ac3722b31f6b1891fc97))

Reply:
Note that #79500 had the same kind of duplicated-ish note messages, [e.g.](https://github.com/rust-lang/rust/pull/79500/files#diff-8917aa6c77e1ed1126ebd8ad4ee5653c8a379b99ba5d43507a82f13df4577a8b).

Reply:
I also don't know the cause here, but it feels like it might be a larger bug -- maybe worth trying to reproduce on stable/nightly outside of the String type (e.g., have a struct and add a generic parameter to it) and see if that causes similar behavior on regular rustc compilers? If so, then you could file an issue for that.

If this is something specific to String that's probably worth poking at too.

Reply:
It does not appear to be specific to String; I have opened #101992 about it.